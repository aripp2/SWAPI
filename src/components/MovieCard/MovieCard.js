import React from 'react';
import './MovieCard.scss';
import { Link } from 'react-router-dom';
//should this be a class component so we can hold on to the charaters property as part of it's state and be able to fetch that data that is only specific to the movie card?

const MovieCard = ({ id, title, date, characters, getCharacters }) => {
  console.log("characters", characters)
  return (
      <div className="movie_card">
        <div className="movie_card-style">
          <h3 className="episode" >Episode {id}</h3>
          <h4 className="movie_card-title">{title}</h4>
          <h5>Released in {date}</h5>
        <Link to='/characters'><button className="movie_card-btn"
        onClick={() => getCharacters(characters)}>View Characters
        </button></Link>
        </div>
      </div>
    )
}

export default MovieCard;