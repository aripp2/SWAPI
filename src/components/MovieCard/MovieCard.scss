@import url('https://fonts.googleapis.com/css?family=Luckiest+Guy&display=swap');
@import url('https://fonts.googleapis.com/css?family=Bitter:400,700&display=swap');

$title-font: 'Luckiest Guy', cursive;
$body-font: 'Bitter', serif;

.movie_card {
  position: relative; 
  box-shadow: 0px 5px 5px 0px #e8e8e8, 0px 10px 10px 0px #e8e8e8;
  margin: 20px; 
  padding: 11px; 
}

h5,
.movie_card-title {
  font-family: $body-font;

}

.movie_card-btn{
    border-color: #7878782b;
    border-radius: 20px;
    font-size: large;
    background-color: #f0ffffba;
    font-weight: 900;
    letter-spacing: 1.1px;
    padding: 12px;
    border: .5px solid #80808021;
}

.episode {
  font-family: $title-font;
  letter-spacing: 1px;
}

