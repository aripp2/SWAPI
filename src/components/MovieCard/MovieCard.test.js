import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard'
import { Link } from 'react-router-dom';

describe('MovieCard',() => { 
  let wrapper;
  const mockCharacters = ["https://swapi.co/api/people/2/", "https://swapi.co/api/people/3/", "https://swapi.co/api/people/10/"]

  beforeEach(() => {
    const mockGetCharacters = jest.fn();
    wrapper = shallow(<MovieCard 
      key={1}
      id={1}
      title={"The Phantom Menace"}
      date={"1999-05-19"}
      characters={mockCharacters}
      getCharacters={mockGetCharacters} />);
  })

  it('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call getCharacters on click', () => {

    wrapper.find('button').simulate('click')

    expect(mockGetCharacters).toHaveBeenCalled()

  })
})



