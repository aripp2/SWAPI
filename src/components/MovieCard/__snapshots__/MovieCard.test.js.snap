// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match Snapshot 1`] = `
<div
  className="movie_card"
>
  <h3
    className="episode"
  >
    Episode 
    I
  </h3>
  <h4
    className="movie_card-title"
  >
    The Phantom Menace
  </h4>
  <h5>
    Released on 
    May 19 1999
  </h5>
  <Link
    to="/movies/1"
  >
    <button
      className="movie_card-btn"
      onClick={[Function]}
    >
      View Characters
    </button>
  </Link>
</div>
`;
